R/compat-defer.R

```diff
@@ -1,14 +1,26 @@
-# nocov start --- compat-defer ---
+# ---------------------------------------------------------------------------------
+# Compatibility file inlined from https://github.com/r-lib/withr/blob/main/R/compat-defer.R
+# Do not edit by hand
+# ---------------------------------------------------------------------------------
+
+# ---
+# source: https://github.com/r-lib/withr/blob/main/R/compat-defer.R
+# last-updated: 2022-03-03
```

Comment:
Is `last-updated` going to be the compat file version? If so we might want to include HH:MM? Just because when I'm working on a compat file I might send multiple updates in a single day.

Not a strong opinion, it's easier to edit if only the date changes. But if we introduce more automatic tools for updating compat files this increases the chance of someone fetching the wrong version by luck.

For instance if:
- An update is pushed to compat-A
- A revdep fetches compat-A
- Another update is pushed to compat-A
- An update is pushed to compat-B adding a dependency on the latest version of compat-A, only the date in this case.
- The revdep fetches compat-B, and the tool doesn't update compat-A because the required version matches.

I guess one way of avoiding this is for the tool to always update all compat files. But sometimes you don't want to be up to date everywhere because updating requires some supervision in case of breaking changes. But it's also reasonable to always update _dependencies_.

So in this case (if we always update deps) we don't even need to add required versions. So I guess I've talked myself out of adding HH:MM.

Reply:
I think the ruler is too long? Here is how it looks for me:

<img width="630" alt="Screenshot 2023-01-19 at 10 14 45" src="https://user-images.githubusercontent.com/4465050/213402483-2b7be38b-90bb-47bb-97f1-bb9c9cda8f13.png">

Reply:
72 characters tend to fit in all screens.

Reply:
Plus we normally review `compat` file edits, so I don't think we're likely to be pushing to master multiple times in a day